ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -37,11 +36,11 @@ StatusDialog {
     QtObject {
         id: d
          property var debounceFetchSuggestedRoutes: Backpressure.debounce(root, 1000, function() {
-                 root.swapAdaptor.fetchSuggestedRoutes(payPanel.rawValue)
+             root.swapAdaptor.fetchSuggestedRoutes(payPanel.rawValue)
          })
 
         function fetchSuggestedRoutes() {
-            if (payPanel.valueValid) {
+            if (payPanel.valueValid && !!payPanel.selectedHoldingId) {
```

Comment:
this should not be needed as check isFormCorrectlyFilled() in the adpator and that should be enough?

Reply:
Yeah not strictly needed, but it imho launches the `loading` animation unnecessarily, even when a token is not selected yet

Reply:
root.swapInputParamsForm.isFormFilledCorrectly() is better in that case, Ive already added that to my PR https://github.com/status-im/status-desktop/pull/15261/files#:~:text=if%20(payPanel.valueValid%20%26%26,root.swapAdaptor.newFetchReset()

Reply:
Yup, I can retro add it here instead :)

Reply:
But then might not be the best idea, given how we want to handle the input/output params

Reply:
what do mean not the best idea? instead of !!payPanel.selectedHoldingId we add isFormCorrectlyFilled() which checks if ether to or from token is set. In case either is set its still valid